### DISCUSSION TITLE
Why Budibase convert the custom sql parameters from v1.0.1xx?

### DISCUSSION THREAD

**Discussion by samg666** [Budibase/budibase#5390] [Ideas]
Hello,

I have #5187 about the date time params converting issue, although it's bee fiexed on v1.0.111, but I still keep having more issues. All issue was due to the change of Budibase handling on sql parameters. 

After studied the MySQL logs, I found that if the params was string then BB will add single quote, if there's special charaters like `%`, then BB will use `concat('%', 'param1', '%') ` to convert it, if the params is a number then no change(I am not sure if it's 100% works like this). 

Beside #5187 issue, I have just found another issue is about custom SQL ordering, I have a parameters named `OrderBy` and I have a dropdown to pass the ordering params something like `id DESC` or `name DESC`. After upgraded to v1.0.1xx, Budibase will convert it to `'id DESC'` which added single quote and make ordering not working now. 

I think there must some reason for Budibase to convert the parameters. But for my understanding, Budibase convert the parameters bring more confusions and less flexibility. It's much difficult to troubleshooting SQL issues as what we saw on the query panel is not what to be sent to the MySQL. I will prefer keep the previous v1.0.9x logic, it's much more flexiable and easier to troubleshot.   

@mike12345567 Mike, appreciate for your help on #5187, could you give some idea on how can I fix the ordering issue and is it possible to keep the query panel code to What-You-See-Is-What-You-Get.

**Comment by mike12345567** [comment]
Hi @ggs331 - we made the change for security reasons, I'm sorry you've had so many issues with it, we've noticed specifically quite a number of issues with it around MySQL (due to type considerations in SQL variables).

If you do find any problems and can provide re-producible examples of them please raise them as issues for us to have a look at, this is a relatively recent change and we're sorry for teething issues with it, but its something that we need to move forward with - in the long run once it is fully smoothed out we believe the change will be better for the platform/gives Budibase a lot more control over parameterisation of queries and allows us to add levels of protection to them.

**Reply by samg666** [reply]
@mike12345567 I have submitted an issue #5395.
